renderer/reducers/pay.js

```diff
@@ -257,18 +259,22 @@ export const queryRoutes = invoice => async (dispatch, getState) => {
           throw error
         }
 
-        // There is no guarentee that the lnd node has the Router service enabled.
+        // There is no guarantee that the lnd node has the Router service enabled.
         // Fall back to using queryRoutes if we got some other type of error.
         routes = await callQueryRoutes()
       }
     }
-
     // For older versions use queryRoutes.
     else {
       routes = await callQueryRoutes()
     }
 
-    dispatch({ type: QUERY_ROUTES_SUCCESS, routes })
+    // Check if routes match current PR to be sure we will be paying the correct PR.
+    if (paymentRequest === paymentRequestSelector(getState())) {
+      dispatch({ type: QUERY_ROUTES_SUCCESS, routes, paymentRequest })
+    } else {
+      mainLog.warn('Received route data for a stale pr')
+    }
   } catch (e) {
     dispatch({ type: QUERY_ROUTES_FAILURE, error: e.message })
```

Comment:
Should we only dispatch this if the query that failed matches the currently active one? What happens if an older query fails whilst a newer one is now running?

Reply:
this is the whole point of this change ? dispatch only if PR and routes match

Reply:
I’m referring to the failure case

Reply:
ahhh, ok. with git it's not clear sometimes

Reply:
If the old one fails after successful one has completed it will erase "good" routes and reset querying state which will cause issues.